pkg/diagnostics/grpc_tracing.go

```diff
@@ -67,6 +61,15 @@ func GRPCTraceUnaryServerInterceptor(appID string, spec config.TracingSpec) grpc
 			}
 		}
 
+		// Check for trace context in response headers
+		if itrnlInvkResp, ok := resp.(internalv1pb.InternalInvokeResponse); ok {
```

Comment:
This could also be runtimev1pb responses as well. 
Also it should be commonv1pb.InvokeResponse instead of internalv1pb.InternalInvokeResponse..

Reply:
Removed the checks as per discussion